packages/orbit-components/src/InputGroup/index.js

```diff
@@ -178,85 +191,123 @@ const InputGroup = ({
   onFocus,
   onBlur,
   onChange,
+  onBlurGroup,
 }: Props): React.Node => {
   const [active, setActive] = React.useState(false);
   const [filled, setFilled] = React.useState(false);
   const inputID = useRandomId();
+  const [tooltipShown, setTooltipShown] = React.useState(false);
+  const [tooltipShownHover, setTooltipShownHover] = React.useState(false);
+  const labelRef = React.useRef(null);
+  const iconRef = React.useRef(null);
+  const foundErrors = React.useMemo(() => findPropInChild("error", children), [children]);
+  const foundHelp = React.useMemo(() => findPropInChild("help", children), [children]);
+
+  const errorReal = error || (foundErrors.length > 0 && foundErrors[0]);
+  const helpReal = help || (foundHelp.length > 0 && foundHelp[0]);
+  const randomId = useRandomIdSeed();
 
   const isFilled = React.useCallback(
-    () =>
-      setFilled(
-        findPropInChild("value", children).length === React.Children.toArray(children).length,
-      ),
+    () => setFilled(findPropInChild("value", children).length > 0),
     [children],
   );
 
   React.useEffect(() => {
     isFilled();
   }, [isFilled]);
 
-  const handleFocus = (ev: SyntheticInputEvent<HTMLInputElement>) => {
+  const handleFocus = (ev: SyntheticInputEvent<HTMLInputElement>, callBack) => {
     setActive(true);
-    if (onFocus) {
-      onFocus(ev);
-    }
+    setTooltipShown(true);
+    if (onFocus) onFocus(ev);
+    if (callBack) callBack(ev);
   };
 
-  const handleBlur = (ev: SyntheticInputEvent<HTMLInputElement>) => {
+  const handleBlur = (ev: SyntheticInputEvent<HTMLInputElement>, callBack) => {
     isFilled();
     setActive(false);
-    if (onBlur) {
-      onBlur(ev);
-    }
+    setTooltipShown(false);
+    if (onBlur) onBlur(ev);
+    if (callBack) callBack(ev);
   };
 
-  const handleChange = (ev: SyntheticInputEvent<HTMLInputElement>) => {
+  const handleChange = (ev: SyntheticInputEvent<HTMLInputElement>, callBack) => {
     isFilled();
-
     if (onChange) {
       onChange(ev);
     }
+    if (callBack) callBack(ev);
+  };
+
+  const handleBlurGroup = ev => {
+    ev.persist();
+    if (onBlurGroup) {
+      setTimeout(() => {
+        setActive(isActive => {
+          if (!isActive) {
+            onBlurGroup(ev);
```

Comment:
What is the purpose of `onBlurGroup`? It would be best to document it in the README.

Reply:
it's inside readme 🤔 . It's left after previous implementation of error forms, i think